### PR TITLE
fix: Ingredient parsing improvement for additives

### LINKED ISSUE #6536: Ingredient parsing improvement for additives
```Emulsifying (INS 471, INS 477) & Stabilizing Agents (INS 412, INS 410)```
is not properly parsing: https://world.openfoodfacts.org/product/8904057300040/chocobar-flavoured-ice-cream-arun

### PR DESCRIPTION
**Pull request by borrokk**
### Checklist

Make sure you've done all the following (You can delete the checklist before submitting)

- [x] Code is well documented
- [x] Include unit tests for new functionality
- [x] Code passes GitHub workflow checks in your branch
- [x] If you have multiple commits please combine them into one commit by squashing them.
- [x] Read and understood the [contribution guidelines](https://github.com/openfoodfacts/openfoodfacts-server/blob/main/CONTRIBUTING.md)

### Description

I have added the word "emulsifying" as a nova:en synonym for emulsifiers in taxonomies/additives_classes.txt file and also added a test in t/ingredients.t 

### Related issue(s) and discussion
- Fixes #6536: Emulsifying (INS 471, INS 477) & Stabilizing Agents (INS 412, INS 410) is not properly parsing

### CODE REVIEW

**Comment by stephanegigandet** [comment]
> I have added the word "emulsifying" as a nova:en synonym for emulsifiers in taxonomies/additives_classes.txt file.

The PR seems to include other files (the logos for fruits et légumes), can you make a separate PR just for this one?

Question: have you tested that it indeed solves the problem? Are you running it locally?

It would be useful to add a test in t/ingredients.t

**Comment by alexgarel** [comment]
@borrok you have to make your title more descriptive (maybe title for bug you mention). Imagine when you do a "git log" or "git blame", you really would like to know what the commit does.

Also you have to put a prefix to your title, in this case "fix:" (we follow https://semantic-gitlog.github.io/semantic-gitlog/#/)

**Comment by borrokk** [comment]
> > I have added the word "emulsifying" as a nova:en synonym for emulsifiers in taxonomies/additives_classes.txt file.
> 
> The PR seems to include other files (the logos for fruits et légumes), can you make a separate PR just for this one?
> 
> Question: have you tested that it indeed solves the problem? Are you running it locally?
> 
> It would be useful to add a test in t/ingredients.t

Yes, I will make the changes. I haven't tested it if it solves the problem, I might need some help with that. I will add a test too.

**Comment by borrokk** [comment]
> @Borrok you have to make your title more descriptive (maybe title for bug you mention). Imagine when you do a "git log" or "git blame", you really would like to know what the commit does.
> 
> Also you have to put a prefix to your title, in this case "fix:" (we follow https://www.conventionalcommits.org/)

Sure, I will make the changes.

**Comment by borrokk** [comment]
should I delete this PR and give a separate one just for this issue? I made the commits to the previous branch so the pr might include other files too.

File: t/ingredients.t [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/perl -w
 
-use Modern::Perl '2017';
+use Modern::Perl '2014';
```
Comment (alexgarel) [inline t/ingredients.t:3]: Why this change ?

**Review by alexgarel — CHANGES_REQUESTED** [review]
It's ok, but can we keep `use Modern::Perl '2017';` ?

File: t/ingredients.t (hunk 1 above)
Comment (borrokk) [reply t/ingredients.t:3]: oh this I was trying to run docker build and I made some changes since I was getting errors, I forgot to undo the change. We can keep 2017.

Comment (borrokk) [reply t/ingredients.t]: It should work with 2017, but I don't know why the Perl unit tests are failing when I change it to 2017.

Comment (alexgarel) [reply t/ingredients.t]: @borrokk I think you forgot to include expected_test_results/ingredients/en-emulsifier-synonyms.json

Look at [log of last test run](https://pipelines.actions.githubusercontent.com/serviceHosts/2177ada6-e616-4c97-899b-f91e94e24e4c/_apis/pipelines/1/runs/63725/signedlogcontent/8?urlExpires=2022-04-25T13%3A44%3A23.9964973Z&urlSigningMethod=HMACV1&urlSignature=eC2X6WTBGvpvnNg1HzAPgBYGDmeIOUYS1%2FTF%2FORaCfM%3D)

> 2022-04-23T07:09:56.3850948Z #   Failed test 'could not load expected_test_results/ingredients/en-emulsifier-synonyms.json'
2022-04-23T07:09:56.3851403Z #   at t/ingredients.t line 455.

Comment (borrokk) [reply t/ingredients.t]: Yes, I will look into this and make the changes.

Comment (borrokk) [reply t/ingredients.t]: I added the json file but still getting errors. May I know what changes I need to make?

**Review by alexgarel — APPROVED** [review]
:tada: kudos @borrokk
